packages/itmat-commons/src/models/user.ts

```diff
@@ -14,6 +14,9 @@ export interface IUserWithoutToken {
     emailNotificationsActivated: boolean;
     deleted: number | null;
     createdBy: string;
+    createdAt: number;
+    expiredAt: number;
+    locked: boolean
```

Comment:
I don't understand how this flag is necessary. i.e. `if currentTime > expiredAt -> locked`

Reply:
exactly

Reply:
does "locked" mean something else? But would be good to have a function where admin can "disable" a user at will, like if a user is transferred to another project for a while (but might come back). would be good to "freeze" their account but can revive it later with all the permission and user info intact. 

But this can be implemented by setting expireAt to 0 when admin "freezes" the account. so might be better to get rid of "locked" flag anyway

Reply:
The reason to set both 'expiredAt' and locked is, based on my understanding, e.g., a project may last 2 months, thus the 'expiredAt' would be 2 months later. But if the admin want to review/modify the projects during this time,  like system maintenance, it would be good to lock all users temporarily such that they would not change anything.

Reply:
if you are looking to lock all users, the lock should be at the app level right? but i dont think we need that right now. 

I like that the expiry date to be linked to a project though; but then we have to implement a expiry date for the project as well. also a user might be linked to multiple project